src/EnergyPlus/OutputReportTabular.cc

```diff
@@ -13254,32 +13238,34 @@ namespace OutputReportTabular {
-                    powerGenRadIntoSurf(kTimeStep) = powerGenRadSeq(desDaySelected, kTimeStep, zoneIndex) * QRadThermInAbsMult;
-                    lightLWRadIntoSurf(kTimeStep) = lightLWRadSeq(desDaySelected, kTimeStep, zoneIndex) * QRadThermInAbsMult;
-                    // for each time step, step back through time and apply decay curve
+                    // for each time step, step back through time and apply decay curve to radiant heat for each end use absorbed in each surface
                     Real64 peopleConvFromSurf = 0.0;
                     Real64 equipConvFromSurf = 0.0;
                     Real64 hvacLossConvFromSurf = 0.0;
                     Real64 powerGenConvFromSurf = 0.0;
                     Real64 lightLWConvFromSurf = 0.0;
                     Real64 lightSWConvFromSurf = 0.0;
                     Real64 feneSolarConvFromSurf = 0.0;
-                    for (int mStepBack = 1; mStepBack <= kTimeStep; ++mStepBack) {
-                        peopleConvFromSurf += peopleRadIntoSurf(kTimeStep - mStepBack + 1) * decayCurve(mStepBack, jSurf);
-                        equipConvFromSurf += equipRadIntoSurf(kTimeStep - mStepBack + 1) * decayCurve(mStepBack, jSurf);
-                        hvacLossConvFromSurf += hvacLossRadIntoSurf(kTimeStep - mStepBack + 1) * decayCurve(mStepBack, jSurf);
-                        powerGenConvFromSurf += powerGenRadIntoSurf(kTimeStep - mStepBack + 1) * decayCurve(mStepBack, jSurf);
-                        lightLWConvFromSurf += lightLWRadIntoSurf(kTimeStep - mStepBack + 1) * decayCurve(mStepBack, jSurf);
+
+                    for (int mStepBack = 1; mStepBack <= NumOfTimeStepInHour * 24; ++mStepBack) {
+                        int sourceStep = kTimeStep - mStepBack + 1;
+                        if (mStepBack > kTimeStep) {
+                            sourceStep = NumOfTimeStepInHour * 24 - (mStepBack - kTimeStep) + 1;
+                        }
+                        assert(sourceStep > 0);
+                        Real64 thisQRadThermInAbsMult = TMULTseq(desDaySelected, sourceStep, radEnclosureNum) *
+                            ITABSFseq(desDaySelected, sourceStep, jSurf) * Surface(jSurf).Area *
+                            decayCurve(mStepBack, jSurf);
+                        peopleConvFromSurf += peopleRadSeq(desDaySelected, sourceStep, zoneIndex) * thisQRadThermInAbsMult;
+                        equipConvFromSurf += equipRadSeq(desDaySelected, sourceStep, zoneIndex) * thisQRadThermInAbsMult;
+                        hvacLossConvFromSurf += hvacLossRadSeq(desDaySelected, sourceStep, zoneIndex) * thisQRadThermInAbsMult;
+                        powerGenConvFromSurf += powerGenRadSeq(desDaySelected, sourceStep, zoneIndex) * thisQRadThermInAbsMult;
+                        lightLWConvFromSurf += lightLWRadSeq(desDaySelected, sourceStep, zoneIndex) * thisQRadThermInAbsMult;
```

Comment:
Accumulate these delayed loads directly from the source data across all of the history terms.

Reply:
I don't think this is right. It is not taking the previous timesteps of the peopleRadIntoSurf as the original code. I think the arrays should be both timestep and surface so the first should be: peopleRadIntoSurf(kTimeStep, jSurf)

Reply:
@JasonGlazer Yes, `peopleRadIntoSurf` could be stored in 2d, I suppose that would save repeated multiplications of TMULTSeq*ITABSFseq*Area. The way it's done here, sourceStep is moving with mStepBack, so it is picking up the previous timesteps of `peopleRadIntoSurf` (by recomputing them every time here). I can redo it that way, but that does prevent accumulating histories prior to midnight, because they won't be computed yet as kTimeStep advances. 

But given your and @rraustad 's reservations about the 24 hr accumulation, I think I should revert that, save output based on this algorithm, then redo it and get the same result (but with fewer computations).

Reply:
These are summed in variable(kTimeStep) arrays at line 13290 and I think that's the intent of the model.

Reply:
@mjwitte I think the only change needed from here is the calculation of decayCurveCool/Heat over a 24 hour period. I think everything would then fall into place.

Reply:
@mjwitte, ok I see what you are saying about eliminating those arrays and I agree. Your new code is cleaner and removes the need for the arrays.

Reply:
@rraustad @JasonGlazer I've reverted the extended 24-hr history, so the decay history is as it was before, from the current `kTimeStep` back to midnight.